src/compiler/utilities.ts

```diff
@@ -7239,8 +7239,8 @@ export function isRightSideOfQualifiedNameOrPropertyAccess(node: Node) {
 
 /** @internal */
 export function isRightSideOfAccessExpression(node: Node) {
-    return isPropertyAccessExpression(node.parent) && node.parent.name === node
-        || isElementAccessExpression(node.parent) && node.parent.argumentExpression === node;
+    return !!node.parent && (isPropertyAccessExpression(node.parent) && node.parent.name === node

```

Comment:
Other utils like this one do not have a guard like this so it looks a little bit off. However, `.parent` is actually nullable - a `SourceFile` doesn't have a parent. I could guard this before calling this function but personally, I prefer to have it here.